cmake/DefaultTargetProperties.cmake

```diff
@@ -58,7 +58,10 @@ function(set_default_target_properties target_name)
 
     # Hide all symbols from external statically linked libraries.
     if (IS_GCC_OR_CLANG)
-        target_link_options(${target_name} PRIVATE -Wl,--exclude-libs,ALL)
+        if (NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
```

Comment:
[consider][minor] We discussed being consistent re `CMAKE_SYSTEM_NAME` vs. `APPLE`

Reply:
Sorry missed that one, doh. Fixed [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/366/commits/b4afe8f0507fbf62299921ad18518c9aa87d69d7).